syft/messaging/plan.py

```diff
@@ -378,6 +378,15 @@ def find_location(self, args):
                     return arg.location
         return sy.framework.hook.local_worker
 
+    def _replace_reference_ids(self, old_id):
+        # Replace ids that reference self.id
+        self.replace_ids(
+            from_ids=self.arg_ids, to_ids=self.arg_ids, from_worker=old_id, to_worker=self.id
+        )
+        self.replace_ids(
+            from_ids=self.result_ids, to_ids=self.result_ids, from_worker=old_id, to_worker=self.id
+        )
```

Comment:
> As from_ids == to_ids and from_worker -> to_worker is identicial in the 2 calls, only one call to replace_ids looks necessary

I just checked `self.replace_ids`, we need both of the calls since the first replate the arg_ids and the second replace the result_ids.

Reply:
> Maybe also clarify the comment: we're updating the worker id to own the operations on the plan

Done